massa-versioning/src/versioning.rs

```diff
@@ -1793,104 +1793,164 @@ mod test {
                 ]),
                 stats: MipStoreStats::new(mip_stats_cfg.clone()),
             };
 
             assert_matches!(
-                vs_raw_1.update_with(&vs_raw_2),
+                store_1.update_with(&store_2),
                 Err(UpdateWithError::Overlapping(..))
             );
-            assert_eq!(vs_raw_1.store.get(&vi_1).unwrap().state, vs_1.state);
-            assert_eq!(vs_raw_1.store.get(&vi_2).unwrap().state, vs_2.state);
+            assert_eq!(store_1.store.get(&mi_1).unwrap().state, ms_1.state);
+            assert_eq!(store_1.store.get(&mi_2).unwrap().state, ms_2.state);
 
             // Check that try_from fails too (because it uses update_with internally)
             {
-                let _vs_raw_2_ = MipStoreRaw::try_from((
+                let _store_2_ = MipStoreRaw::try_from((
                     [
-                        (vi_1.clone(), vs_1.clone()),
-                        (vi_2_2.clone(), vs_2_2.clone()),
+                        (mi_1.clone(), ms_1.clone()),
+                        (mi_2_2.clone(), ms_2_2.clone()),
                     ],
                     mip_stats_cfg.clone(),
                 ));
-                assert_eq!(_vs_raw_2_.is_err(), true);
+                assert_eq!(_store_2_.is_err(), true);
             }
         }
 
-        // case 2
+        // case 2: overlapping versioning component
         {
-            let mut vs_raw_1 = MipStoreRaw::try_from((
-                [(vi_1.clone(), vs_1.clone()), (vi_2.clone(), vs_2.clone())],
+            let mut store_1 = MipStoreRaw::try_from((
+                [(mi_1.clone(), ms_1.clone()), (mi_2.clone(), ms_2.clone())],
                 mip_stats_cfg.clone(),
             ))
             .unwrap();
 
-            let mut vi_2_2 = vi_2.clone();
-            vi_2_2.components = vi_1.components.clone();
+            let mut mi_2_2 = mi_2.clone();
+            // Make MIP invalid (has component version set to 1 - same as MIP 1)
+            mi_2_2.components = mi_1.components.clone();
 
-            let vs_2_2 = advance_state_until(ComponentState::defined(), &vi_2_2);
-            let vs_raw_2 = MipStoreRaw {
+            let ms_2_2 = advance_state_until(ComponentState::defined(), &mi_2_2);
+            let store_2 = MipStoreRaw {
                 store: BTreeMap::from([
-                    (vi_1.clone(), vs_1.clone()),
-                    (vi_2_2.clone(), vs_2_2.clone()),
+                    (mi_1.clone(), ms_1.clone()),
+                    (mi_2_2.clone(), ms_2_2.clone()),
                 ]),
                 stats: MipStoreStats::new(mip_stats_cfg.clone()),
             };
 
             // MIP-0003 in vs_raw_1 & vs_raw_2 has != components
             assert_matches!(
-                vs_raw_1.update_with(&vs_raw_2),
+                store_1.update_with(&store_2),
+                Err(UpdateWithError::Overlapping(..))
+            );
+        }
+
+        // case 3: trying to downgrade network version
```

Comment:
You test with equal versions, so the "downgrade" comment is not perfect

Reply:
Done